doc/esmvalcore/preprocessor.rst

```diff
@@ -303,6 +303,19 @@ is for example useful for climate models which do not offer land/sea fraction
 files. This arguments also accepts the special dataset specifiers
 ``reference_dataset`` and ``alternative_dataset``.
 
+Optionally you can specify your own custom fx variable to be used in cases when e.g. a certain
+experiment is preferred for fx data retrieval:
+
+.. code-block:: yaml
+
+    preprocessors:
+      preproc_weighting:
+        weighting_landsea_fraction:
+          area_type: land
+          exclude: ['CanESM2', 'reference_dataset']
+          fx_files: [{'short_name': 'sftlf', 'exp': 'piControl'}, {'short_name': 'sftof', 'exp': 'piControl'}]
```

Comment:
Wouldn't it be better to use the keyword `fx_variables` instead of `fx_files`? The current name suggests you have to give  it a path to a certain file which might be confusing (see for example https://esmvaltool.readthedocs.io/projects/esmvalcore/en/latest/recipe/preprocessor.html#extract-shape with the `shapefile` keyword.

Reply:
Also this looks like it could be kept more readable by using the same format as for data sets:
```
datasets:
  - {dataset: CanESM2, project: CMIP5, exp: historical, ensemble: r1i1p1, start_year: 2001, end_year: 2004}
  - {dataset: UKESM1-0-LL, project: CMIP6, exp: historical, ensemble: r1i1p1f2, start_year: 2001, end_year: 2004, grid: gn}
  - {dataset: EC-EARTH3, alias: custom_alias, project: CMIP6, exp: historical, ensemble: r1i1p1f1, start_year: 2001, end_year: 2004, grid: gn}
```

Reply:
replaced with `fx_variables` -> the user can specify whatever keys they want and in whatever order they want them

Reply:
> the user can specify whatever keys they want and in whatever order they want them

sorry for not being clear, I didn't mean the should add these specific keys and order, just the way it is formatted, so instead of the current:
`fx_files: [{'short_name': 'sftlf', 'exp': 'piControl'}, {'short_name': 'sftof', 'exp': 'piControl'}]`
have the other format in the documentation: 
```
fx_files: 
  - {'short_name': 'sftlf', 'exp': 'piControl'}
  - {'short_name': 'sftof', 'exp': 'piControl'}
```
But it's just a suggestion. Polishing nail extensions.

Reply:
Also I saw you now replaced the argument `fx_variables` on lots of other places as well, but the docstrings still say that these `fx_variables` then should be a dictionary with `{field: filename of fx variable}`. There's also some other places where still the name `fx_files` is used. maybe check whether in these instances that you now changed it's really not the full filename but just the variable name? If you change these couple, then change the other occurrences as well. (e.g `def tile_grid_areas(cube, fx_files):`)

Reply:
cheers mate @JaroCamphuijsen I'll have a look&correct tomorrow, it is :pizza: time now :grin:

Reply:
Hi @valeriupredoi how was the pizza? Did you have a look at this already?

Reply:
great pizza, cheers, I can't remember the toppings though :grin: So here's the thing: the way the user can ask for `fx_variables` in the recipe (as arg of the preprocessor step) can be either as a list of short names eg `fx_variables: ['sftlf', 'sftof']` or as a list of dicts eg `fx_variables: [{short_name: sftlf, mip: fx}, {short_name: sftof, mip: fx}]` so making the listing look like the variable dictionary will mean that the user will be confused as to why it could also be just a simple list; `tile_grid_areas` actually takes as argument the fx files and not the variables, so that's why I left it with `fx_files` in

Reply:
good catch though - I just realized I missed adding documentation for these funcs (now added) :beer:

Reply:
Great, but....
for example line 159 in esmvalcore/preprocessor/_area.py 
`def tile_grid_areas(cube, fx_files):`

And 'fx_files' is used on more places in these files. 

Sorry to be an ass....

Reply:
Cool, if you think this won't cause confusion I'm perfectly fine with this!

Reply:
should be fine, RTFM will play a good part here :grin: